<launch>

 <arg name="rviz_path" default="$(find nuturtle_robot)/config/waypoint.rviz" doc="path to load the rviz configuration file from" />

 <arg name="robot" default="1" doc="The turtle bot number to run"/>
 <include file="$(find nuturtle_robot)/launch/basic_remote.launch">
 <arg name="robot" value="$(arg robot)"/>
 </include> 


<!-- rosparam param="waypoint_x">[2.0, 5.0, 5.92, 3.50, 1.08]</rosparam -->
<!-- rosparam param="waypoint_y">[1.0, 1.0, 2.85, 4.61, 2.85]</rosparam -->

<!-- rosparam param="waypoint_x">[0.20, 0.5, 0.592, 0.350, 0.108]</rosparam -->
<!-- rosparam param="waypoint_y">[0.10, 0.1, 0.285, 0.461, 0.285]</rosparam -->

<rosparam param="waypoint_x">[0.80, 2.0, 2.368, 0.40, 0.432]</rosparam>
<rosparam param="waypoint_y">[0.40, 0.4, 1.14, 1.844, 1.14]</rosparam>
<param name="frequency" value="100"/>
<param name="tolerance" value="0.065"/>

  <rosparam command="load" file="$(find nuturtle_description)/config/diff_params.yaml" />
  <param name="robot_description" command="xacro '$(find nuturtle_gazebo)/urdf/diff_drive.gazebo.xacro'"/>

<node name="waypoint" pkg="nuturtle_robot" type="nuturtle_robot_waypoint" output="screen">
<param name="frac_vel" value="0.5"/>
<param name="px" value="0.5"/>
<param name="pw" value="0.2"/>
</node>

<node pkg="nuturtle_robot" type="nuturtle_robot_turtle_interface_node" name="turtle_interface">
<param name="left_wheel_joint" value="left_wheel_axle"/>
<param name="right_wheel_joint" value="right_wheel_axle"/>
</node>



<node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher"/>

<node machine="turtlebot" pkg="rigid2d" type="rigid2d_odometry_node"  name="fake_odometry" ns="fake">
<param name="odom_frame_id" value="odom"/>
<param name="body_frame_id" value="base_link"/>
</node>


<node machine="turtlebot" pkg="rigid2d" type="rigid2d_fake_encoders" name="fake_encoders">
<remap from="/joint_states" to="/fake/joint_states"/>
</node>

<node machine="turtlebot" pkg="rigid2d" type="rigid2d_odometry_node"  name="odometry">
<param name="odom_frame_id" value="odom"/>
<param name="body_frame_id" value="base_link"/>
</node>


<node machine="turtlebot" pkg="tf2_ros" type="static_transform_publisher" name="fake_odom_broadcaster" args="0 0 0 0 0 0 1 odom fake_odom" />
<node name="rviz" pkg="rviz" type="rviz" args="-d $(arg rviz_path)" if="$(eval robot != 0)" />

</launch>
